jgrapht-guava/src/main/java/org/jgrapht/graph/guava/MutableValueGraphAdapter.java

```diff
@@ -0,0 +1,270 @@
+/*
+ * (C) Copyright 2018-2018, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.graph.guava;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import java.util.Objects;
+import java.util.function.ToDoubleFunction;
+
+import org.jgrapht.Graph;
+import org.jgrapht.GraphType;
+import org.jgrapht.util.TypeUtil;
+
+import com.google.common.graph.EndpointPair;
+import com.google.common.graph.Graphs;
+import com.google.common.graph.MutableValueGraph;
+import com.google.common.graph.ValueGraph;
+import com.google.common.graph.ValueGraphBuilder;
+
+/**
+ * A graph adapter class using Guava's {@link MutableValueGraph}.
+ *
+ * <p>
+ * The class uses a converter from Guava's values to JGraphT's double weights. Thus, the resulting
+ * graph is weighted. This is a one-way conversion meaning that calling
+ * {@link #setEdgeWeight(EndpointPair, double)} will throw an unsupported operation exception.
+ * Adjusting the weights can be done directly (by keeping an external reference) on the underlying
```

Comment:
can you be a little more explicit here (perhaps a small example)? I'm not sure that it is very obvious how to change the weight of an edge.